airbyte-cdk/python/unit_tests/sources/streams/concurrent/scenarios/stream_facade_builder.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import logging
+from typing import Any, List, Mapping, Optional, Tuple, Union
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog, ConnectorSpecification, DestinationSyncMode, SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.message import InMemoryMessageRepository, MessageRepository
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.concurrent.adapters import StreamFacade
+from airbyte_protocol.models import ConfiguredAirbyteStream
+from unit_tests.sources.file_based.scenarios.scenario_builder import SourceBuilder
+
+
+class StreamFacadeSource(AbstractSource):
```

Comment:
Note to @maxi297: There will be changes needed on the incremental development to support incremental for this